scripts/appS3.py

```diff
@@ -0,0 +1,103 @@
+import os
+import string
+import time
+import random
+import boto3
+import json
+import argparse
+
+# seed random
+random.seed(time.time())
+
+
+def app_handle(args):
+    keys_path = args.keys
+    endpoint = args.endpoint
+    if not os.path.isfile(keys_path):
+        print("Provided key path {} is incorrect.".format(keys_path))
+        return
+
+    # Fetch Auth
+    with open(keys_path, 'r') as keys_file:
+        keys_dict = json.load(keys_file)
+
+    # Create Boto3 Client
+    keys = keys_dict["keys"][0]
+    client = boto3.resource("s3", verify=False,
+                            endpoint_url=endpoint,
+                            aws_access_key_id=keys["access_key"],
+                            aws_secret_access_key=keys["secret_key"])
+
+    # Perform IO
+    objects = []
+    bucket_name = "test-bucket"
+    client.Bucket(bucket_name).create()
+    for i in range(args.obj_num):
+        object_name = "test-object"+rand_str(4)
+        data = str(rand_str(random.randint(10, 30)))*1024*1024
+        primary_object_one = client.Object(
+            bucket_name,
+            object_name
+        )
+        primary_object_one.put(Body=data)
+        # Store for
+        objects.append(
+            (object_name, primary_object_one.content_length/(1024*1024))
+        )
+
+    # Print Summary
+    print("IO Summary: Object Count {}".format(args.obj_num))
+    for obj, size in objects:
+        print("Object: {}/{} -> Size: {}MB".format(bucket_name, obj, size))
+
+    # Cleanup (if asked for)
+    if not args.no_delete:
+        print("Performing Cleanup")
+        for obj, size in objects:
+            client.Object(bucket_name, obj).delete()
+
+
+def rand_str(length: int):
+    return "".join(
+        random.choices(string.ascii_uppercase + string.digits, k=length)
+    )
+
+
+if __name__ == "__main__":
+    argparse = argparse.ArgumentParser(
+        description="An application which uses S3 for storage",
+        epilog="Ex: python3 app.py --keys keys.txt",
+    )
+
+    argparse.add_argument(
+        "endpoint",
+        type=str,
+        help="Provide RGW endpoint to talk to.",
+    )
+    argparse.add_argument(
+        "keys",
+        type=str,
+        help="Provide JSON file generated from Ceph RGW Admin.",
+    )
+    argparse.add_argument(
+        "--obj-num",
+        type=int,
+        default=1,
+        help="Number of objects to upload to S3.",
+    )
+    argparse.add_argument(
+        "--no-delete",
+        type=bool,
+        default=False,
```

Comment:
For boolean arguments, I think it's best to set `action="store_true"` so that both the default is set and also so that the user doesn't need to pass a value.

Reply:
i think default=False does the same, for the execution used on the github action:
```
ubuntu@microcephtest:~$ python3 script.py http://localhost:80 test.json --obj-num 2
Namespace(endpoint='http://localhost:80', keys='test.json', obj_num=2, no_delete=False, func=<function app_handle at 0x7f1015eefd90>)
```

Reply:
@lmlg thoughts ?

Reply:
@UtkarshBhatthere The default=False is subtly differently. IT makes it a flag on the commandline versus an argument that expects a value if it is specified. In the case where you have wish to pass `--no-delete` you must specify it like `foo.py --no-delete True ...` when in reality all you want is the `--no-delete` as a flag on the commandline.

The 'store_true' option referenced by @lmlg does that.

Reply:
Ack, Thank you for the explanation @wolsen 🙇🏼